src/blas/KokkosBlas3_gemm_inner_fix.hpp

```diff
@@ -0,0 +1,166 @@
+
+  KOKKOS_INLINE_FUNCTION
+  InnerGemmFixC(const int as0, const int as1, const int bs0, const int bs1,
+                const int cs0, const int cs1)
+      : _as0(as0), _as1(as1), _bs0(bs0), _bs1(bs1), _cs0(cs0), _cs1(cs1) {}
+
+  // serial rank update
+  template <typename ScalarType, typename ValueType>
+  KOKKOS_INLINE_FUNCTION int serial_invoke(const ScalarType alpha,
+                                           const ValueType *KOKKOS_RESTRICT A,
+                                           const ValueType *KOKKOS_RESTRICT B,
+                                           const int k,
+                                           /**/ ValueType *KOKKOS_RESTRICT C);
+
+  // serial rank update for remainder
+  template <typename ScalarType, typename ValueType>
+  KOKKOS_INLINE_FUNCTION int serial_invoke(const ScalarType alpha,
+                                           const ValueType *KOKKOS_RESTRICT A,
+                                           const ValueType *KOKKOS_RESTRICT B,
+                                           const int m, const int k,
+                                           /**/ ValueType *KOKKOS_RESTRICT C);
+
+  // serial rank update for remainder
+  template <typename ScalarType, typename ValueType>
+  KOKKOS_INLINE_FUNCTION int serial_invoke(const ScalarType alpha,
+                                           const ValueType *KOKKOS_RESTRICT A,
+                                           const ValueType *KOKKOS_RESTRICT B,
+                                           const int m, const int n,
+                                           const int k,
+                                           /**/ ValueType *KOKKOS_RESTRICT C);
+
+  template <typename MemberType, typename ScalarType, typename ValueType>
+  KOKKOS_INLINE_FUNCTION int team_invoke(const MemberType &member,
+                                         const ScalarType alpha,
+                                         const ValueType *KOKKOS_RESTRICT A,
+                                         const ValueType *KOKKOS_RESTRICT B,
+                                         const int k,
+                                         /**/ ValueType *KOKKOS_RESTRICT C);
+
+  // team rank update for remainder
+  template <typename MemberType, typename ScalarType, typename ValueType>
+  KOKKOS_INLINE_FUNCTION int team_invoke(const MemberType &member,
+                                         const ScalarType alpha,
+                                         const ValueType *KOKKOS_RESTRICT A,
+                                         const ValueType *KOKKOS_RESTRICT B,
+                                         const int m, const int n, const int k,
+                                         /**/ ValueType *KOKKOS_RESTRICT C);
+};
+
+}  // namespace KokkosBlas
+
+#include "KokkosBlas3_serial_gemm_inner_fixa_impl.hpp"
+// TODO: fix compilation errors InnerGemmFixB (not used internally, not tested)
+// #include "KokkosBlas3_serial_gemm_inner_fixb_impl.hpp"
```

Comment:
Compiler errors in this header were not detected before, because it was not used anywhere (no internal need for `InnerGemmFixB` and no tests for it).